File: ThreatModelDfdService/Data/DTO/CreateDfdElementDTO.cs
using ThreatModelDfdService.Model.Enums;

namespace ThreatModelDfdService.Data.DTO;

public record CreateDfdElementDTO(
    string ElementName,
    DfdElementType Type,
    decimal PositionX,
    decimal PositionY,
    decimal Width,
    decimal Height,
    string Uuid
);

File: dfd-service/src/main/java/com/tcc/dfd_service/dto/CreateDfdElementDTO.java
package com.tcc.dfd_service.dto;

import com.tcc.dfd_service.enums.DfdElementType;
import java.math.BigDecimal;

public record CreateDfdElementDTO(
        String elementName,
        DfdElementType type,
        BigDecimal positionX,
        BigDecimal positionY,
        BigDecimal width,
        BigDecimal height,
        String uuid
) {
}